### PR TITLE
Replacing Peewee by SQLAlchemy + Alembic

### PR DESCRIPTION
**Pull request by MiltonLn**

### CODE REVIEW

File: backend/src/side_stacker/repositories.py [hunk 1]
```diff
@@ -4,15 +4,16 @@
...
 
 
 class AbstractRepository(abc.ABC):
     def __init__(self):
-        self.game = None
+        self.game: Dict = dict()
 
     def assert_present(self):
-        assert self.game is not None, "is not supposed to be called without creating a game first"
+        assert self.game != dict(), "is not supposed to be called without creating a game first"
```
Comment (yamijuan) [inline backend/src/side_stacker/repositories.py:16]: hmmm I kinda prefer `assert self.game`, I think this should be enough

File: backend/src/side_stacker/repositories.py [hunk 2]
```diff
@@ -37,63 +38,53 @@ def commit(self):
...
-    def _serialize_moves(moves: List[Tuple]) -> str:
-        return json.dumps(moves)
-
-    @staticmethod
-    def _parse_moves(moves: str) -> List[Tuple]:
-        return json.loads(moves)
-
     def find(self, game_key: str, refresh: bool = False) -> Optional[dict]:
-        if self.game is None or refresh:
+        if self.game == dict() or refresh:
```
Comment (yamijuan) [inline backend/src/side_stacker/repositories.py:53]: `if not self.game or refresh:` should be enough right?

File: backend/src/tests/test_game_repository.py [hunk 3]
```diff
@@ -28,41 +22,33 @@ def do_test_create(game_repository):
...
-
-
-def do_test_find(mock__find, game_repository, id=None):
     mock__find.return_value = None
     GAME_KEY = "12345"
     expected_game = {"key": GAME_KEY, "board": [], "players": []}
 
     # Without game
-    assert game_repository.find(GAME_KEY) is None
+    assert game_repository.find(GAME_KEY) == dict()
```
Comment (yamijuan) [inline backend/src/tests/test_game_repository.py:30]: same comment just for consistency

**Review by yamijuan — APPROVED** [review]

File: backend/src/side_stacker/repositories.py (hunk 1 above)
Comment (MiltonLn) [reply backend/src/side_stacker/repositories.py:16]: Well, an empty dict's truthy value is False so this makes sense, we can implement it that way

**Pull request merged**